src/util/codec/mysql/json/mod.rs

```diff
@@ -47,3 +53,77 @@ pub enum Json {
     Boolean(bool),
     None,
 }
+
+// https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-array
+pub fn json_array(elems: Vec<Datum>) -> Result<Json> {
+    let mut a = Vec::with_capacity(elems.len());
+    for elem in elems {
+        a.push(try!(elem.into_json()));
+    }
+    Ok(Json::Array(a))
+}
+
+// https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-object
+pub fn json_object(kvs: Vec<Datum>) -> Result<Json> {
+    let len = kvs.len();
+    if !is_even(len) {
+        return Err(Error::Other(box_err!("Incorrect parameter count in the call to native \
+                                          function 'JSON_OBJECT'")));
+    }
+    let mut map = BTreeMap::new();
+    let mut key = None;
+    for elem in kvs.into_iter() {
+        if key.is_none() {
+            // take elem as key
+            if elem == Datum::Null {
+                return Err(invalid_type!("JSON documents may not contain NULL member names"));
+            }
+            key = Some(try!(elem.into_string()));
+        } else {
```

Comment:
you can use `let keys, vals =kvs.into_iter().enumerate().partition(|(&n, &m)| n & 1 == 0)`